optimum/exporters/onnx/utils.py

```diff
@@ -310,6 +310,38 @@ def get_stable_diffusion_models_for_export(
     return models_for_export
 
 
+def _get_submodels_for_export_sam(model, variant):
+    models_for_export = {}
+
+    if variant == "monolith":
+        models_for_export["model"] = model
+    else:
+        # We use the model patcher to patch their forward method.
+        models_for_export["vision_encoder"] = model
+        models_for_export["prompt_mask"] = model
```

Comment:
Or, is `prompt_mask` a combination of the `prompt_encoder` and `mask_decoder`?

Reply:
Yes, maybe I should name it more explicitly `prompt_encoder_mask_decoder`?

Reply:
That could help I guess (albeit a bit verbose 😅).

Also, something you've probably thought about already: is there a situation where the user may want to use the prompt encoder and mask decoder separately? I haven't needed to worry about this (yet), but just asking in case.

Reply:
I guess when we want to encode a point for several different images?

Reply:
I'll leave as is and just rename to `prompt_encoder_mask_decoder.onnx.`

Reply:
> I guess when we want to encode a point for several different images?

Indeed, quite a rare use case 😅 

> I'll leave as is and just rename to prompt_encoder_mask_decoder.onnx.

Sounds good!